### ISSUE TITLE
ArgumentNullException when there are 0 SectionHeaders

### ISSUE THREAD

**Issue by ZonderP** [secana/PeNet#93]
If there are no SectionHeaders then the PeFile constructor throws a System.ArgumentNullException in
DataDirectoryParsers.cs in the constructor of internal class DataDirectoryParsers when trying to call _sectionHeaders = sectionHeaders.ToArray(); because 'sectionHeaders' is null because the passed in 'ImageSectionHeaders' is null.
In NativeStructureParsers.cs after call to 'InitNTHeadersParser' from within 'InitAllParsers' the
ImageNtHeaders.FileHeader.NumberOfSection = 0.

This happens for me with the following file:
C:\Program Files (x86)\Adobe\Acrobat Reader DC\Reader\icucnv40.dll

New to GitHub, so I'm not sure if this is the expected way to report issues...

**Comment by secana** [comment]
Hi @ZonderP 
thx for reporting this. I had a look at the Adobe file. While it has an *.dll extension it is not actually not a Windows PE file. It does not start with an MZ Header nor has any other parsable PE header parts in it. 
In that case there is not much I can do. To avoid running in such an error, you can check a file with `PeNet.PeFile.IsPEFile(...)`, before trying to parse it.

**Comment by ZonderP** [comment]
Oh my!
Sorry I took your time, I was not aware of 'IsPEFile'.
I already tried renaming quite some files to 'dll' when I started using your fantastic utility to see how PeFile constructor behaves and immediately found out that I would need to catch 'IndexOutOfBounds'. This helped for all those files. Thus I thought 'icucnv40.dll' actually would be a 'special' PE file with no sections or what.
Thanks again - will from now on use 'IsPEFile'.
